tests/ecosystem/upgrade/conftest.py

```diff
@@ -0,0 +1,276 @@
+    ):
+        rbd_pods = create_pods(
+            interface=constants.CEPHBLOCKPOOL,
+            pvc_factory=pvc_factory,
+            pod_factory=pod_factory,
+            storageclass=default_storageclasses.get(reclaim_policy)[0],
+            count=1,
+            access_mode=constants.ACCESS_MODE_RWO,
+            volume_mode=constants.VOLUME_MODE_FILESYSTEM,
+        )
+        pods.extend(rbd_pods)
+
+        for access_mode in (
+            constants.ACCESS_MODE_RWO,
+            constants.ACCESS_MODE_RWX
+        ):
+            cephfs_pods = create_pods(
+                interface=constants.CEPHFILESYSTEM,
+                pvc_factory=pvc_factory,
+                pod_factory=pod_factory,
+                storageclass=default_storageclasses.get(reclaim_policy)[1],
+                count=1,
+                access_mode=access_mode,
+            )
+            pods.extend(cephfs_pods)
+
+    return pods
+
+
+@pytest.fixture
+def post_upgrade_block_pods(
+    pvc_factory,
+    pod_factory,
+    default_storageclasses
+):
+    """
+    Generate RBD pods for tests after upgrade is executed.
+    These pods use block volume type.
+
+    Returns:
+        list: List of pods with RBD interface
+    """
+    pods = []
+    for reclaim_policy in (
+        constants.RECLAIM_POLICY_DELETE,
+        constants.RECLAIM_POLICY_RETAIN
+    ):
+        for access_mode in (
+            constants.ACCESS_MODE_RWX,
+            constants.ACCESS_MODE_RWO
+        ):
+            rbd_pods = create_pods(
+                interface=constants.CEPHBLOCKPOOL,
+                pvc_factory=pvc_factory,
+                pod_factory=pod_factory,
+                storageclass=default_storageclasses.get(reclaim_policy)[0],
+                count=1,
+                access_mode=access_mode,
+                volume_mode=constants.VOLUME_MODE_BLOCK,
+            )
+            pods.extend(rbd_pods)
+
+    return pods
+
+
+@pytest.fixture(scope='session')
+def upgrade_fio_file(tmp_path_factory):
+    """
+    File that controls the state of running fio on pods during upgrade.
+    """
+    upgrade_fio_file = tmp_path_factory.mktemp('upgrade_testing')
+    upgrade_fio_file = upgrade_fio_file.joinpath('fio_status')
+    upgrade_fio_file.write_text('running')
+    return upgrade_fio_file
+
+
+@pytest.fixture(scope='session')
+def pre_upgrade_pods_running_io(
+    pre_upgrade_filesystem_pods,
+    pre_upgrade_block_pods,
+    upgrade_fio_file
+):
+
+    def run_io_in_bg():
+        """
+        Run IO by executing FIO and deleting the file created for FIO on
+        the pod, in a while true loop. Will be running as long as
+        the upgrade_fio_file contains string 'running'.
+        """
+        while upgrade_fio_file.read_text() == 'running':
+            for pod in pre_upgrade_filesystem_pods:
+                log.warning(f"Running fio on fs pod {pod.name}")
+                pod.run_io(
+                    storage_type='fs',
+                    size='1GB'
```

Comment:
@fbalak how do we make sure FIO keeps running for extended period?  Don't we need the **runtime** param  ? Ignore if it's already handled and I misunderstood.

Reply:
`run_io` has a default value of runtime set to 60 seconds and the function is triggered repeatedly.